tests/unit/test_replay_manager.py

```diff
@@ -1,94 +1,261 @@
+    {
+        "simconfig_fixture": "ringtest_baseconfig",
+        "extra_config": {
+            "inputs": {
+                "spikeReplay" : {
+                    "module": "synapse_replay",
+                    "input_type": "spikes",
+                    "spike_file": INPUT_SPIKES_FILE,
+                    "delay": 0,
+                    "duration": 50,
+                    "node_set": "RingB",
+                }
+            }
+        }
+    },
+], indirect=True)
 @pytest.mark.forked
-def test_replay_stim_generated(replay_sim_config_file):
+def test_replay_stim_generated_run(create_tmp_simulation_config_file):
     from neurodamus import Neurodamus
     from neurodamus.core import NeurodamusCore as Nd
     from neurodamus.core.configuration import Feature
     from neurodamus.connection import NetConType
 
-    n = Neurodamus(
-        replay_sim_config_file.name,
+    nd = Neurodamus(
+        create_tmp_simulation_config_file,
         restrict_features=[Feature.Replay],
         disable_reports=True,
         cleanup_atexit=False,
     )
 
-    edges_a_b = n.circuits.get_edge_manager("RingA", "RingB")
+    edges_a_b = nd.circuits.get_edge_manager("RingA", "RingB")
     conn_1_1001 = next(edges_a_b.get_connections(1001, 1))
     time_vec = conn_1_1001._replay.time_vec.as_numpy()
     npt.assert_allclose(time_vec, [0.1, 2.275, 4.35, 7.725])
 
     # get the netcon for connection 1->1001
-    cell = n._pc.gid2cell(1001)
-    pre_cell = n._pc.gid2cell(1)
+    cell = nd._pc.gid2cell(1001)
+    pre_cell = nd._pc.gid2cell(1)
     nclist = Nd.cvode.netconlist(pre_cell, cell, "")
     assert nclist.count() == 1
     netcon1 = nclist.o(0)
     assert netcon1.weight[4] == int(NetConType.NC_PRESYN)
 
     # get the netcon generated by ReplayStim for connection 1->1001
     nclist = Nd.cvode.netconlist("", cell, "")
+    replay_netcons = []
     for netcon in nclist:
-        if netcon.srcgid() < 0:
-            assert netcon.weight[4] == int(NetConType.NC_REPLAY)
+        if netcon.srcgid() < 0 and netcon.weight[4] == int(NetConType.NC_REPLAY):
+            replay_netcons += [netcon]
             assert netcon.weight[0] == netcon1.weight[0]
             assert netcon.delay == netcon1.delay
             assert netcon.threshold == 10
+    assert len(replay_netcons) > 0
+
+    # get voltage variations in cell 1001
+    cell = nd.circuits.get_node_manager("RingB").get_cell(1001)
+    voltage_vec = Nd.Vector()
+    voltage_vec.record(cell._cellref.soma[0](0.5)._ref_v)
+
+    Nd.finitialize()
+    nd.run()
+
+    v_increase_rate = np.diff(voltage_vec, 2)
+    window_sum = np.convolve(v_increase_rate, [1, 2, 4, 2, 1], 'valid')
+    strong_reduction_pos = np.nonzero(window_sum < -0.5)[0]
+    non_consecutives_pos = strong_reduction_pos[np.insert(
+        np.diff(strong_reduction_pos) > 1, 0, True)]
+    expected_positions = np.array([32, 75])
+
+    npt.assert_equal(non_consecutives_pos, expected_positions)
```

Comment:
what about putting this in `tests/utils.py`? It is used in various tests. I also have a PR about it if you want to wait

Reply:
I also see [here](https://nrn.readthedocs.io/en/8.2.6/tutorials/ball-and-stick-4.html) that neuron has already a spike detector. Maybe we can use that

Reply:
apparently it works only if you have netcons attached to the cell and it is a rudimentary threshold crossing detector. Maybe it is better to have this function instead.